sdv/metadata/single_table.py

```diff
@@ -307,15 +340,81 @@ def set_sequence_index(self, column_name):
             column_name (str):
                 Name of the sequence index column.
         """
-        if not isinstance(column_name, str):
-            raise ValueError("'sequence_index' must be a string.")
+        self._validate_sequence_index(column_name)
+        self._sequence_index = column_name
+
+    def _validate_sequence_index_not_in_sequence_key(self):
+        """Check that ``_sequence_index`` and ``_sequence_key`` don't overlap."""
+        sk = self._sequence_key
+        sequence_key = {sk} if isinstance(sk, str) else set(sk)
```

Comment:
minor: I think we've been trying to avoid letter abbreviations for variables. If `sequence_key` fits I would say use that instead of sk. Idk if the lint will complain about that later as well